src/lib/y2storage/proposal/autoinst_space_maker.rb

```diff
@@ -72,11 +72,27 @@ def delete_stuff(devicegraph, disk, drive_spec)
 
         # TODO: resizing of partitions
 
-        case drive_spec.use
+        delete_by_use(devicegraph, disk, drive_spec.use)
+      end
+
+      # Deletes unwanted partition according to the "use" element
+      #
+      # @param devicegraph [Devicegraph]
+      # @param disk        [Disk]
+      # @param use         [String,Array<Integer>] Use value ("all", "linux", "free"
+      #   or a list of partition numbers)
+      def delete_by_use(devicegraph, disk, use)
+        return if use == "free"
+        case use
         when "all"
           disk.partition_table.remove_descendants if disk.partition_table
```

Comment:
Any reason to not delete the partition table?

Reply:
That's a fair question. From the AutoYaST point of view, these are different use cases. If you want to wipe the partition table, you set the `initialize` element in the profile; if you only want to use the whole disk (but keeping the partition table if it exists), you set `use=all` (this case).

Reply:
Ok, then your code does what it is expected :). Thanks.